File: csharp/GameOfLife.Engine/Size.cs
namespace GameOfLife.Engine
{
    public class Size
    {
        public Size(int width, int height)
        {
            Width = width;
            Height = height;
        }

        public static Size None => new Size(0, 0);

        public int Width { get; }

        public int Height { get; }

        public override bool Equals(object obj) => 
            obj is Size size
                && size.Width == Width
                && size.Height == Height;

        public override int GetHashCode() => Height ^ Width;
    }
}

File: fsharp/GameOfLife.FSharp.Engine/Size.fs
namespace GameOfLife.FSharp.Engine
    
type Size = {
        width: int;
        height: int;
    } with

    static member none = {
        width = 0;
        height = 0;
    }

module Size =

    let create width height = {
        width = width;
        height = height;
    }
